packages/edit-site/src/components/sidebar-dataviews/default-views.js

```diff
@@ -54,103 +58,114 @@ const DEFAULT_PAGE_BASE = {
 	},
 };
 
-export const DEFAULT_VIEWS = {
-	page: [
-		{
-			title: __( 'All pages' ),
-			slug: 'all',
-			icon: pages,
-			view: DEFAULT_PAGE_BASE,
+export function useDefaultViews( { postType } ) {
+	const labels = useSelect(
+		( select ) => {
+			const { getPostType } = select( coreStore );
+			return getPostType( postType )?.labels;
 		},
-		{
-			title: __( 'Published' ),
-			slug: 'published',
-			icon: published,
-			view: {
-				...DEFAULT_PAGE_BASE,
-				filters: [
-					{
-						field: 'status',
-						operator: OPERATOR_IS_ANY,
-						value: 'publish',
+		[ postType ]
+	);
+	return useMemo( () => {
+		return {
+			[ postType ]: [
+				{
+					title: labels?.all_items || __( 'All items' ),
+					slug: 'all',
+					icon: pages,
+					view: DEFAULT_POST_BASE,
+				},
+				{
+					title: __( 'Published' ),
+					slug: 'published',
+					icon: published,
+					view: {
+						...DEFAULT_POST_BASE,
+						filters: [
+							{
+								field: 'status',
+								operator: OPERATOR_IS_ANY,
+								value: 'publish',
+							},
+						],
 					},
-				],
-			},
-		},
-		{
-			title: __( 'Scheduled' ),
-			slug: 'future',
-			icon: scheduled,
-			view: {
-				...DEFAULT_PAGE_BASE,
-				filters: [
-					{
-						field: 'status',
-						operator: OPERATOR_IS_ANY,
-						value: 'future',
+				},
+				{
+					title: __( 'Scheduled' ),
```

Comment:
We are repeating the status view definition all the time I guess we can do something like:

```
const createStatusView = (title, slug, icon, status) => ({
  title,
  slug,
  icon,
  view: {
    ...DEFAULT_POST_BASE,
    filters: [
      {
        field: 'status',
        operator: OPERATOR_IS_ANY,
        value: status,
      },
    ],
  },
});

// Then use it like:
createStatusView(__('Published'), 'published', published, 'publish'),
createStatusView(__('Drafts'), 'drafts', drafts, 'draft'),
// ... and so on
```

Reply:
I don't think we gain much for now to do this change. Maybe see how to best refactor this code when we work more on `views`.